src/app/cpe/cpe.component.ts

```diff
@@ -1,15 +1,20 @@
-import { Component, OnInit } from '@angular/core';
+import { Component } from '@angular/core';
+import {PageStructure} from "./blue-boxes/model/page-data-structure";
+import {PageStructureService} from "./blue-boxes/services/page-structure.service";
+import {PageStructureValidatorService} from "./blue-boxes/validator/page-structure-validator.service";
 
 @Component({
   selector: 'app-cpe',
   templateUrl: './cpe.component.html',
   styleUrls: ['./cpe.component.scss']
 })
-export class CpeComponent implements OnInit {
+export class CpeComponent {
+  data: PageStructure;
+  constructor(private pageStructureService: PageStructureService, private validatorService: PageStructureValidatorService) {
```

Comment:
I prefer using underscores in front of private variables/parameters (not sure if this is your code as I see it's been moved here)

Reply:
Not my code.

Reply:
But changed accordingly.

Reply:
Prefixed all private services of all components by an underscore.